app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx

```diff
@@ -1,41 +1,70 @@
-import React, { useMemo } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 
 import { ThemeProvider } from "@devtools-ds/themes";
 import { ObjectInspector } from "@devtools-ds/object-inspector";
 import { NetworkLogProperty } from "./NetworkLogProperty";
 
 type Props = {
   label?: string;
-  payload: any;
+  payload?: any;
   isPayloadTooLarge?: boolean;
+  fetchPayload?: () => Promise<any>;
 };
 
-export const NetworkPayload: React.FC<Props> = ({ label, payload }) => {
-  const parsedPayload = useMemo(() => {
-    if (typeof payload === "string") {
-      try {
-        return JSON.parse(payload);
-      } catch (e) {
-        // skip
+export const NetworkPayload: React.FC<Props> = ({ label, payload, fetchPayload }) => {
+  const [fetching, setFetching] = useState(false);
+  const [finalPayload, setFinalPayload] = useState<any>();
+  useEffect(() => {
+    if (payload) {
+      if (typeof payload === "string") {
+        try {
+          return JSON.parse(payload);
+        } catch (e) {
+          // skip
+        }
       }
+      setFinalPayload(payload);
+    } else if (fetchPayload && typeof fetchPayload === "function") {
+      setFetching(true);
+      fetchPayload()
+        .then((fetchedPayload) => {
+          if (typeof fetchedPayload === "string") {
+            try {
+              setFinalPayload(JSON.parse(fetchedPayload));
+            } catch (e) {
+              setFinalPayload(fetchedPayload);
+            }
+          } else {
+            setFinalPayload(fetchedPayload);
+          }
+        })
+        .catch((err) => {
+          console.error(err);
+        })
+        .finally(() => {
+          setFetching(false);
+        });
```

Comment:
`RQNetworkTable` should know the currently selected log ie its parent component should know the selected log and there we should fetch the payload for the selected log and just update the logs state.